llvm/cmake/modules/TableGen.cmake

```diff
@@ -92,8 +92,23 @@ function(tablegen project ofn)
   get_directory_property(tblgen_includes INCLUDE_DIRECTORIES)
   list(TRANSFORM tblgen_includes PREPEND -I)
 
+  set(tablegen_exe ${${project}_TABLEGEN_EXE})
+  set(tablegen_depends ${${project}_TABLEGEN_TARGET} ${tablegen_exe})
```

Comment:
Why the dependency on the tablegen target?  Seems ... recursive?

Reply:
I'm just following what existing code was doing, see comments earlier that start with
> We need both _TABLEGEN_TARGET and _TABLEGEN_EXE in the  DEPENDS list

Reply:
I'd suggest splitting this change (creating two variables and using them in existing places) into a prep commit so it's clear what the function change is.